ffi/capi/src/utf.rs

```diff
@@ -4,33 +4,57 @@
 
 use alloc::borrow::Cow;
 
-use writeable::{impl_display_with_writeable, LengthHint, Writeable};
+use core::fmt::Write;
+use writeable::{LengthHint, Part, TryWriteable, Writeable};
 
-use core::fmt;
+pub(crate) struct LossyWrap<T>(pub T);
+
+impl<T: TryWriteable> Writeable for LossyWrap<T> {
```

Comment:
Note: We might want to move this wrapper into the `writeable` crate if we need it in more places.